src/main/windows/callsWidgetWindow.ts

```diff
@@ -333,50 +306,39 @@ export class CallsWidgetWindow {
      * IPC HANDLERS
      ************************/
 
-    private handleResize = (ev: IpcMainEvent, _: string, msg: CallsWidgetResizeMessage) => {
-        log.debug('onResize', msg);
+    private handleResize = (ev: IpcMainEvent, width: number, height: number) => {
+        log.debug('handleResize', width, height);
 
         if (!this.win) {
             return;
         }
 
-        if (!this.isAllowedEvent(ev)) {
-            log.warn('onResize', 'Disallowed calls event');
+        if (!this.isCallsWidget(ev.sender.id)) {
+            log.debug('handleResize', 'Disallowed calls event');
             return;
         }
 
         const zoomFactor = this.win.webContents.getZoomFactor();
         const currBounds = this.win.getBounds();
         const newBounds = {
             x: currBounds.x,
-            y: currBounds.y - (Math.ceil(msg.height * zoomFactor) - currBounds.height),
-            width: Math.ceil(msg.width * zoomFactor),
-            height: Math.ceil(msg.height * zoomFactor),
+            y: currBounds.y - (Math.ceil(height * zoomFactor) - currBounds.height),
+            width: Math.ceil(width * zoomFactor),
+            height: Math.ceil(height * zoomFactor),
         };
 
         this.setBounds(newBounds);
     }
 
-    private handleShareScreen = (ev: IpcMainEvent, _: string, message: CallsWidgetShareScreenMessage) => {
-        log.debug('handleShareScreen');
+    private handleShareScreen = (ev: IpcMainEvent, sourceID: string, withAudio: boolean) => {
+        log.debug('handleShareScreen', {sourceID, withAudio});
 
-        if (!this.isAllowedEvent(ev)) {
-            log.warn('Disallowed calls event');
+        if (this.mainView?.webContentsId !== ev.sender.id) {
```

Comment:
I decided that it made sense to be more deliberately restrictive on some of these calls methods, so I've removed and reworked all the checks for one-way message passing between the two. It also was necessary to do to merge the preload scripts as the message passing code was getting confused.